hazelcast/src/main/java/com/hazelcast/internal/server/tcp/SingleProtocolDecoder.java

```diff
@@ -99,12 +99,19 @@ public HandlerStatus onRead() {
 
             verifyProtocol(loadProtocol());
             encoder.signalProtocolVerified();
-            // initialize the connection
+
+            // Initialize the connection
             initConnection();
             setupNextDecoder();
-
-            if (shouldSignalProtocolLoaded()) {
-                ((MemberProtocolEncoder) encoder.getFirstOutboundHandler()).signalProtocolLoaded();
+            if (!channel.isClientMode()) {
+                // Set up the next encoder in the pipeline if  in client mode
```

Comment:
did you wanted to say "Set up the next encoder in the pipeline if  **not** in client mode"

Reply:
Fixed it.